### PR TITLE
Flex 2758 convert remove device fitnesse tests to cucumber

### PR DESCRIPTION
**Pull request by SamVanDijk**
There is a bug in the code so that removing a device isn't possible. The bug is reported in FLEX-2035.

### CODE REVIEW

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/database/core/FirmwareSteps.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+     *
+     * @param expectedEntity
+     * @throws Throwable
+     */
+    @Then("^the entity firmware exists$")
+    public void theEntityFirmwareExists(final Map<String, String> expectedEntity) throws Throwable {
+        // TODO: Wait until the stuff is created.
+        final Firmware entity = this.firmwareRepo.findByFilename(getString(expectedEntity, Keys.FIRMWARE_FILENAME));
+
+        // Assert.assertEquals(getString(expectedEntity, "ManufacturerId",
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/database/core/FirmwareSteps.java]: Why isn't this not checked?

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-core/FirmwareManagement/UpdateFirmware.feature [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    When receiving an update firmware request
+      | FirmwareFilename         | NewFirmware          |
+      | FirmwarePushToNewDevices | false                |
+      | Manufacturer             | Test                 |
+      | ModelCode                | TestModel            |
+      | FirmwareDescription      | Firmware is changed! |
+    Then the update firmware response contains
+      | Result | OK |
+
+  Scenario: Update firmware with an unknown devicemodel
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-core/FirmwareManagement/UpdateFirmware.feature]: Combine this and the scenario below

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/database/core/FirmwareSteps.java (hunk 1 above)
Comment (SamVanDijk) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/database/core/FirmwareSteps.java]: This is changed in FLEX-2687 the FirmwareManagement branch

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-core/FirmwareManagement/UpdateFirmware.feature (hunk 2 above)
Comment (SamVanDijk) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-core/FirmwareManagement/UpdateFirmware.feature]: This is changed in FLEX-2687 the FirmwareManagement branch

**Pull request merged**